packages/web_benchmarks/lib/analysis.dart

```diff
@@ -48,22 +49,36 @@ BenchmarkResults computeDelta(
     }
 
     final List<BenchmarkScore> testScores = test.scores[benchmarkName]!;
-    for (final BenchmarkScore score in testScores) {
+    final List<BenchmarkScore> deltaScores = delta.scores[benchmarkName]!;
+
+    for (int i = 0; i < testScores.length; i++) {
+      final BenchmarkScore testScore = testScores[i];
       // Lookup this metric in the baseline.
       final BenchmarkScore? baselineScore = baselineScores
-          .firstWhereOrNull((BenchmarkScore s) => s.metric == score.metric);
+          .firstWhereOrNull((BenchmarkScore s) => s.metric == testScore.metric);
       if (baselineScore == null) {
         continue;
       }
 
-      // Add the delta to the [testMetric].
-      score.delta = (score.value - baselineScore.value).toDouble();
+      final BenchmarkScore scoreWithDelta = BenchmarkScore(
+        metric: testScore.metric,
+        value: testScore.value,
+        delta: (testScore.value - baselineScore.value).toDouble(),
+      );
+      deltaScores
+        ..removeAt(i)
+        ..insert(i, scoreWithDelta);
```

Comment:
This still treats `BenchmarkResults` as mutable, and the reason this works is due to the misfortune that in Dart lists and maps are mutable by default. Can we create a new `Map<String, List<BenchmarkScore>> scoreDeltas`, populate it with deltas and then `return BencharkResults(scoreDeltas)`?

Reply:
Okay switched to building a map and then parsing into a new BenchmarkResults object. I also add `growable: false` to the construction of each scores list so that the lists are actually immutable.